lib/vdc/DataMgr.cpp

```diff
@@ -3131,58 +3148,83 @@ int DataMgr::_getVar(size_t ts, string varname, int level, int lod, float *data)
     return (0);
 }
 
-int DataMgr::_getLatlonExtents(string varname, bool lonflag, float &min, float &max, long ts)
+void DataMgr::_getLonExtents(vector<float> &lons, vector<size_t> dims, float &min, float &max) const
 {
+    min = max = 0.0;
+    VAssert(dims.size() == 1 || dims.size() == 2);
+
+    if (!lons.size()) return;
+
+    size_t nx = dims[0];
+    size_t ny = dims.size() > 1 ? dims[1] : 1;
+    for (size_t j = 0; j < ny; j++) {
+        GeoUtil::UnwrapLongitude(lons.begin() + (j * nx), lons.begin() + (j * nx) + nx);
+        GeoUtil::ShiftLon(lons.begin() + (j * nx), lons.begin() + (j * nx) + nx);
+    }
+    min = *(std::min_element(lons.begin(), lons.end()));
+    max = *(std::max_element(lons.begin(), lons.end()));
```

Comment:
It seems you can use `std::minmax_element()` to avoid one loop of the array.
https://en.cppreference.com/w/cpp/algorithm/minmax_element

The same applies to code 20 lines down.

Reply:
Done.